Automattic/WP-Job-Manager
# Fix search by query string with empty categories

Fixes #1977

#### Changes proposed in this Pull Request:

The goal of this PR was to fix the search by query string, when passing slugs of empty categories.
There were 2 possible approaches.

1) One approach is a cleaner solution, but I think the UX is worse. Basically, the first option is to fetch the results based on the query string if the categories filter in the form is empty.  See: https://github.com/Automattic/WP-Job-Manager/commit/6bde71f04eba276795e4a70808dc56cccb02d452
2) The other approach (the current code of this branch), is to add the categories from the query string to the form, and as a consequence, the result is filtered correctly considering also the query string. It's better in terms of UX because the user has the reference of what is being filtered in the form and can go forth and back any moment.

#### Testing instructions:

* Create some jobs and add them to categories.
* Create one category to be empty.
* In the jobs page, add the query string: `?search_category=my-empty-category` with the slug of the empty category. The result should be empty.
* Try to filter by categories with results and make sure that the results are correct.
* Go to the `wp-admin`
* Go to  `Job Listings` → `Settings`
* Active the option `Multi-select Categories`
* Try to filter per multiple categories using the query string: `?search_category=my-empty-category,my-other-category` (separate per comma)

#### Note:

The `search_category` for multiple categories is only supported if the `Multi-select Categories` is activated.
